cnudie/util.py

```diff
@@ -414,6 +417,140 @@ def find_changed_component(
+):
+    if delivery_dashboard_url_view_diff:
+        bom_diff_header = f'## <a href="{delivery_dashboard_url_view_diff}">BoM Diff</a>\n'
+    else:
+        bom_diff_header = '## BoM Diff\n'
+
+    added_components = [
+        f'\U00002795 {component.name} {component.version}'
+        for component in component_diff.cidentities_only_right
+        if component.name not in component_diff.names_version_changed
+    ]
+
+    removed_components = [
+        f'\U00002796 {component.name} {component.version}'
+        for component in component_diff.cidentities_only_left
+        if component.name not in component_diff.names_version_changed
+    ]
+
+    changed_components = [
+        f'\U00002699 {new_component.name}: {old_component.version} → {new_component.version}'
+        for old_component, new_component in component_diff.cpairs_version_changed
+    ]
+
+    summary_counts = textwrap.dedent(f'''\
+        Added components: {len(added_components)}
+        Changed components: {len(changed_components)}
+        Removed components: {len(removed_components)}\n
+    ''')
+
+    summary_details = []
+    if added_components:
+        summary_details.append('### Added Components:\n' + '\n'.join(added_components) + '\n')
+
+    if removed_components:
+        summary_details.append('### Removed Components:\n' + '\n'.join(removed_components) + '\n')
+
+    if changed_components:
+        summary_details.append('### Changed Components:\n' + '\n'.join(changed_components) + '\n')
+
+    component_details = []
+    for old_component, new_component in component_diff.cpairs_version_changed:
+        component_name = f'{new_component.name}'
+        if delivery_dashboard_url:
+            component_link = (
+                f"<a href='{delivery_dashboard_url}/#/component?name={new_component.name}'>"
+                f'{component_name}</a>'
+            )
+        else:
+            component_link = component_name
+
+        component_header = (
+            f'<details><summary>\U00002699 {component_link}:'
+            f'{old_component.version} → {new_component.version}</summary>\n')
+
+        added_resources = []
+        removed_resources = []
+        changed_resources = []
+
+        # Group resources by name
+        old_resources_grouped = {}
+        new_resources_grouped = {}
+
+        for res in old_component.resources:
+            old_resources_grouped.setdefault(res.name, []).append(res)
+
+        for res in new_component.resources:
+            new_resources_grouped.setdefault(res.name, []).append(res)
+
+        # Process each resource in the new component
+        for res_name, new_res_list in new_resources_grouped.items():
+            old_res_list = old_resources_grouped.get(res_name, [])
+
+            if (
+                old_res_list
+                and sorted([res.version for res in old_res_list])
+                == sorted([res.version for res in new_res_list])
+            ):
+                # skip
```

Comment:
It's obvious that the iteration is skipped here because of the following `continue` keyword, rather state _why_ it is skipped here :-)